arm/Microsoft.MachineLearningServices/workspaces/computes/deploy.bicep

```
@@ -0,0 +1,118 @@
+param location string = resourceGroup().location
+
+@description('Required. Specifies the sku, also referred as "edition".')
+@allowed([
+  'Basic'
+  'Enterprise'
+])
+param sku string
+
+@description('Optional. Identity for the resource.')
+param identity object = {}
+
+@description('Optional. Contains resource tags defined as key/value pairs.')
+param tags object = {}
+
+@description('Required. Flag to specify whether to deploy the compute. Necessary as the compute resource is not idempontent, i.e. a second deployment will fail. Therefore, this flag needs to be set to "false" as long as the compute resource exists.')
+param deployCompute bool
+
+@description('Optional. Location for the underlying compute.')
+param computeLocation string = resourceGroup().location
+
+@description('Optional. The description of the Machine Learning compute.')
+param computeDescription string = ''
+
+@description('Optional. Opt-out of local authentication and ensure customers can use only MSI and AAD exclusively for authentication.')
+param computeDisableLocalAuth bool = false
+
+@description('Optional. ARM resource id of the underlying compute.')
+param computeResourceId string = ''
+
+@description('Required. Set the object type.')
+@allowed([
+  'AKS'
+  'AmlCompute'
+  'ComputeInstance'
+  'Databricks'
+  'DataFactory'
+  'DataLakeAnalytics'
+  'HDInsight'
+  'Kubernetes'
+  'SynapseSpark'
+  'VirtualMachine'
+])
+param computeType string
+
+@description('Optional. The properties of the compute. Will be ignored in case "computeResourceId" is set.')
+param computeProperties object = {}
+
+@description('Optional. Enable telemetry via the Customer Usage Attribution ID (GUID).')
+param enableDefaultTelemetry bool = true
+
+// =========== //
+// Existing resources //
+// =========== //
+resource machineLearningWorkspace 'Microsoft.MachineLearningServices/workspaces@2021-04-01' existing = {
+  name: machineLearningWorkspaceName
+}
+
+// =========== //
+// Deployments //
+// =========== //
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource machineLearningWorkspaceCompute 'Microsoft.MachineLearningServices/workspaces/computes@2022-01-01-preview' = if (deployCompute == true) {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: sku
+    tier: sku
+  }
+  parent: machineLearningWorkspace
+  identity: identity
+  properties: union({
+    computeLocation: computeLocation
+    description: computeDescription
+    disableLocalAuth: computeDisableLocalAuth
+    computeType: computeType
+  }, (!empty(computeResourceId) ? {
+    resourceId: computeResourceId
```

If you'd agree we should try to stay aligned with the Azure resource reference. The rationale is, that it is (in theory) easier for consumers, as our readMe would be closer aligned to the offical Azure Resource Reference. If we diverge from it, there should be a good reason. 
> Note: This comment also relates to other properties like `computeDescription`

```suggestion
    resourceId: resourceId
```

---

@MrMCake Updated. "description" is not possible -> "This expression is referencing its own declaration, which is not allowed.", therefore I use "computeDescription".

---

@dr-dolittle actually it is. Use the annotation `sys.description()` instead. We have the same in case of e.g. the resources in the `Microsoft.Authorization` provider namespace

---

@MrMCake Is updated.